### PR TITLE
Add `Page.browser()` method

### PR DESCRIPTION
**Pull request by arnemart**
As suggested in #2275, having access to the browser instance from the page object can be useful. This PR implements that.

This is my first PR and I'm not intimately familiar with the architecture of puppeteer, so I'm not sure if this is the best way of implementing the feature, but feel free to show me the error of my ways if I'm doing something stupid.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -2553,6 +2561,12 @@ Contains the URL of the response.
 
 ### class: Target
 
+#### target.browser()
+
+- returns: <[Browser]>
+
+Get the browser instance for the associated with the target.
```
Comment (vsemozhetbyt) [inline docs/api.md]: A tiny nit: "...instance ~~for the~~ associated..."

File: docs/api.md [hunk 2]
```diff
@@ -713,6 +715,12 @@ To disable authentication, pass `null`.
 
 Brings page to front (activates tab).
 
+#### page.browser()
+
+- returns: <[Browser]>
+
+Get the browser instance for the page.
```
Comment (aslushnikov) [inline docs/api.md]: nit:
> Get the browser the page belongs to

File: docs/api.md [hunk 3]
```diff
@@ -2553,6 +2561,12 @@ Contains the URL of the response.
 
 ### class: Target
 
+#### target.browser()
+
+- returns: <[Browser]>
+
+Get the browser instance associated with the target.
```
Comment (aslushnikov) [inline docs/api.md]: nit:
> Get the browser the target belongs to

**Review by aslushnikov — APPROVED** [review]

**Comment by KBPratap** [comment]
I tried using API page.browser() and got following error.

```
UnhandledPromiseRejectionWarning: TypeError: page.browser is not a function
(node:77765) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catchblock, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:77765) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

My Environment 
- node v8.11.1
- npm 6.0.0
- running against Google Chrome | 66.0.3359.139
- puppeteer: ^1.3.0

**Comment by aslushnikov** [comment]
@KBPratap it's not released yet. You can install tip-of-tree puppeteer with with the following command:

```
npm install puppeteer@next
```

Also, the next release is scheduled for tomorrow, so you'll be able to get it in pptr 1.4.0.

**Pull request merged**